tests/e2e/specs/flinkArtifact.spec.ts

```diff
@@ -27,37 +28,104 @@ test.describe("Flink Artifacts", { tag: [Tag.CCloud, Tag.FlinkArtifacts] }, () =
-    await artifactsView.deleteFlinkArtifact(uploadedArtifactName);
+  for (const config of entrypoints) {
+    test(config.testName, async ({ page, electronApp }) => {
+      const artifactsView = new FlinkDatabaseView(page);
+      await artifactsView.ensureExpanded();
+      await artifactsView.loadArtifacts(config.entrypoint);
+      const uploadedArtifactName = await startUploadFlow(
+        config.entrypoint,
+        page,
+        electronApp,
+        artifactsView,
+      );
+      await expect(artifactsView.artifacts.filter({ hasText: uploadedArtifactName })).toHaveCount(
+        1,
+      );
 
-    await expect(artifactsView.artifacts.filter({ hasText: uploadedArtifactName })).toHaveCount(0);
-  });
+      await artifactsView.deleteFlinkArtifact(uploadedArtifactName);
+      await expect(artifactsView.artifacts.filter({ hasText: uploadedArtifactName })).toHaveCount(
+        0,
+      );
+    });
+  }
 
-  test("should upload Flink Artifact when cluster selected from the Resources view", async ({
-    page,
-    electronApp,
-  }) => {
-    const artifactsView = new FlinkDatabaseView(page);
-    await artifactsView.loadArtifacts(SelectFlinkDatabase.DatabaseFromResourcesView);
-    await artifactsView.clickSwitchToFlinkResource(FlinkViewMode.Artifacts);
-    const uploadedArtifactName = await artifactsView.uploadFlinkArtifact(electronApp, artifactPath);
+  async function startUploadFlow(
+    entrypoint: SelectFlinkDatabase,
+    page: Page,
+    electronApp: ElectronApplication,
+    artifactsView: FlinkDatabaseView,
+  ): Promise<string> {
+    switch (entrypoint) {
+      case SelectFlinkDatabase.DatabaseFromResourcesView:
+        return await completeArtifactUploadFlow(
+          { entrypoint: SelectFlinkDatabase.DatabaseFromResourcesView },
+          page,
+          electronApp,
+          artifactPath,
+          artifactsView,
+        );
+      case SelectFlinkDatabase.FromArtifactsViewButton:
+        return await completeArtifactUploadFlow(
+          { entrypoint: SelectFlinkDatabase.FromArtifactsViewButton },
+          page,
+          electronApp,
+          artifactPath,
+          artifactsView,
+        );
+      case SelectFlinkDatabase.ComputePoolFromResourcesView:
+        return await completeUploadFlowForComputePool(page, electronApp, artifactsView);
+    }
+  }
 
-    await expect(artifactsView.artifacts.filter({ hasText: uploadedArtifactName })).toHaveCount(1);
+  async function completeUploadFlowForComputePool(
+    page: Page,
+    electronApp: ElectronApplication,
+    artifactsView: FlinkDatabaseView,
+  ): Promise<string> {
+    const uploadedArtifactName = await artifactsView.uploadFlinkArtifact(
+      electronApp,
+      artifactPath,
+      true,
+    );
 
-    // Clean up: delete the uploaded artifact using the correct name
-    await artifactsView.deleteFlinkArtifact(uploadedArtifactName);
+    await artifactsView.clickSelectKafkaClusterAsFlinkDatabase();
+    await page.keyboard.type("azure");
```

Comment:
We can't associate compute pools with clusters even under the same env, we should match on provider/region. But how can we be sure there's even a match if we're looking dynamically? 

Also we can't search for cluster via cluster label (has to be the name) so we're stuck if we go down that route. 

I think a dedicated test env, or maybe even org, could start helping out a lot.